Troubleshooting: A drop-down editor does not show data (#4481)
DevExpress/devextreme-documentation: concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/20 AspNet.Data CreateStore Processing.md

> If you call the [DevExtreme.AspNet.Data](https://github.com/DevExpress/DevExtreme.AspNet.Data#devextreme-aspnet-data).[createStore()](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference) method to bind a standalone editor or DataGrid/TreeList lookup column, the editor expects all data operations (including value selection) to be performed on the server. Use the [DevExtreme.AspNet.Data](https://github.com/DevExpress/DevExtreme.AspNet.Data#devextreme-aspnet-data) library to obtain correct data in the [Controller](https://docs.devexpress.com/AspNetCore/401020/devextreme-based-controls/concepts/bind-controls-to-data/api-controllers) for .NET-based applications. If you have another back-end type, adjust your server logic to return data in the proper format according to the [loadOptions](/Documentation/ApiReference/Data_Layer/CustomStore/LoadOptions/) parameters. Refer to the following article for more information: [Server-Side Data Processing](Documentation/Guide/Data_Binding/Specify_a_Data_Source/Custom_Data_Sources/#Load_Data/Server-Side_Data_Processing).

Reviewer: "have" - is a vague/weak verb, so I changed to "work with"

"proper format according to loadOptions" - seems repetitive. If you just say "according to loadOptions", you would automatically imply some sort of a proper format. 

```suggestion
If you call the [DevExtreme.AspNet.Data](https://github.com/DevExpress/DevExtreme.AspNet.Data#devextreme-aspnet-data).[createStore()](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference) method to bind a standalone editor or DataGrid/TreeList lookup column, the editor expects all data operations (including value selection) to be performed on the server. Use the [DevExtreme.AspNet.Data](https://github.com/DevExpress/DevExtreme.AspNet.Data#devextreme-aspnet-data) library to obtain correct data in the [Controller](https://docs.devexpress.com/AspNetCore/401020/devextreme-based-controls/concepts/bind-controls-to-data/api-controllers) for .NET-based applications. If you work with another back-end type, adjust your server logic to return data according to the [loadOptions](/Documentation/ApiReference/Data_Layer/CustomStore/LoadOptions/) parameters. Refer to the following article for more information: [Server-Side Data Processing](Documentation/Guide/Data_Binding/Specify_a_Data_Source/Custom_Data_Sources/#Load_Data/Server-Side_Data_Processing).
```

Examine your network requests to learn which [loadOptions](/Documentation/ApiReference/Data_Layer/CustomStore/LoadOptions/) parameters you need to specify. In the example below the parameter is `filter`.

![DataGrid Lookup Network Request Parameters](/images/Troubleshooting/datagrid-lookup-network-request-parameters.png)

> if your back-end does not support server operations, set the [createStore()](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference) [loadMode](/Documentation/ApiReference/Data_Layer/CustomStore/Configuration/#loadMode) option to `raw` to process data on a client.

Reviewer: ```suggestion
If your back-end does not support server operations, set the [createStore()](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference) [loadMode](/Documentation/ApiReference/Data_Layer/CustomStore/Configuration/#loadMode) option to `raw` to process data on a client.
```

> If you call the [DevExtreme.AspNet.Data](https://github.com/DevExpress/DevExtreme.AspNet.Data#devextreme-aspnet-data).[createStore()](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference) method to bind a standalone editor or DataGrid/TreeList lookup column, the editor expects all data operations (including value selection) to be performed on the server. Set the [createStore()](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference) [loadMode](/Documentation/ApiReference/Data_Layer/CustomStore/Configuration/#loadMode) option to `raw` to process data on a client.

Reviewer 2: loadMode: "raw" won't be efficient if a drop-down editor has large data

I think we should also mention that:

In this case, unutilized our [AspNet.Data](https://github.com/DevExpress/DevExtreme.AspNet.Data#devextreme-aspnet-data) library to obtain correct data in the Controller: [API Controllers](https://docs.devexpress.com/AspNetCore/401020/devextreme-based-controls/concepts/bind-controls-to-data/api-controllers) for .NET-based applications. If you have another back-end type adjust your server logic to  return data in the proper format according to `loadOptions` values: [Server-Side_Data_Processing](https://js.devexpress.com/Documentation/Guide/Data_Binding/Specify_a_Data_Source/Custom_Data_Sources/#Load_Data/Server-Side_Data_Processing)

Only then mention about loadMode e.g.

if your backend doesn't support server operations, set loadMode....

I would also recommend adding a screenshot to demonstrate what we're talking about like I did  in the original topic. It makes text more clear because visual representation is quite important here as well.

> If you use the [DevExtreme.AspNet.Data](https://github.com/DevExpress/DevExtreme.AspNet.Data#devextreme-aspnet-data).[createStore()](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference) method to bind a standalone editor or DataGrid/TreeList lookup column, the editor expects all data operations (including value selection) to be performed on the server. Set the [createStore()](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference) [loadMode](/Documentation/ApiReference/Data_Layer/CustomStore/Configuration/#loadMode) option to `raw` to automatically process data on a client-side.

Reviewer 3: ```suggestion
If you call the [DevExtreme.AspNet.Data](https://github.com/DevExpress/DevExtreme.AspNet.Data#devextreme-aspnet-data).[createStore()](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference) method to bind a standalone editor or DataGrid/TreeList lookup column, the editor expects all data operations (including value selection) to be performed on the server. Set the [createStore()](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference) [loadMode](/Documentation/ApiReference/Data_Layer/CustomStore/Configuration/#loadMode) option to `raw` to process data on a client.
```
